ConfigurationSystem/Client/LocalConfiguration.py

```diff
@@ -255,13 +257,24 @@ def __parseCommandLine(self):
       gLogger.fatal("Error when parsing command line arguments: %s" % str(x))
       self.showHelp(exitCode=2)
 
-    for o, _ in opts:
-      if o in ('-h', '--help'):
+    for opt, val in opts:
+      if opt in ('-h', '--help'):
         self.showHelp()
         sys.exit(2)
+      if opt == '--cfg':
+        self.cliAdditionalCFGFiles.append(val)
 
-    self.cliAdditionalCFGFiles = [arg for arg in args if arg[-4:] == ".cfg"]
-    self.commandArgList = [arg for arg in args if not arg[-4:] == ".cfg"]
+    # environment variable to ensure smooth transition
+    if os.getenv("DIRACNOCFG", None):
+      self.commandArgList = args
+    else:
+      # to avoid issuing the warning for correctly passed cfg files
+      extraCfg = [arg for arg in args if arg[-4:] == ".cfg"]
+      self.cliAdditionalCFGFiles.extend(extraCfg)
+      self.commandArgList = [arg for arg in args if not arg[-4:] == ".cfg"]
+      if extraCfg:
+        # use error level to make sure users will always see the Warning
+        gLogger.error("WARNING: Passing cfg files directly is deprecated. Use --cfg instead.")
```

Comment:
This warning is going to be printed also when people are calling dirac-dms-add-file with a cfg file, right?
With the addition of the env variable they could also just set DIRAC_NO_CFG in their own session or just for the command, right?

Reply:
Also should `gLogger.always` be used instead of `gLogger.error`?

Reply:
@andresailer That was the intention.
@chrisburr We found warning wasn't enough, so we used error. Doesn't .always mean you can't suppress it at all ? That might be one step too far.

Reply:
As there are two questions in my post, I am not sure which was the intention, that the warning is printer when calling dirac-dms-add-file, or that people add the variable in their session, or both.

But I would then propose to update the message to mention the export of the environment variable.

Reply:
@andresailer: The message was very generic as it affects other commands besides dirac-dms-add-file. We were hoping that way we just catch everywhere its used.

Reply:
> We found warning wasn't enough, so we used error. Doesn't .always mean you can't suppress it at all ? That might be one step too far.

Error feels misleading to me as I think it should mean that something has wrong. If someone wants to suppress the message they should fix what they're doing and use `--cfg`.

(apologies for having merged two separate points into this thread)

Reply:
I just mean that for users it is still confusing if they try to add (remove, download...) a .cfg file and get this warning, which doesn't offer them the solution they probably need, as far as I understand.

Reply:
```suggestion
                 gLogger.error("""WARNING: Parsing of '.cfg' files as command line arguments is changing!
Set the environment variable 'export DIRAC_NO_CFG=1' to pass the file as a positional argument. (This will become the default)
To modify the local configuration use '--cfg <configfile>' instead.""")
```
(better indentation of the code could be done with https://docs.python.org/2/library/textwrap.html#textwrap.dedent)

In addition, we could  consider setting the variable inside some of the scripts directly.